### PR TITLE
Concat adjacent strings.

### LINKED ISSUE #553: Adjacent strings should be concatenated
```proto
  option experimental_java_builder_interface =
      "com.google.protobuf.ExtraInterfaces.HasStringValueBuilder"
      "<Proto1, Builder>";
```
From: https://github.com/google/protobuf/blob/b13874d59e976371a1a87e6dc2bf347ed0a0ce5d/java/core/src/test/proto/com/google/protobuf/test_extra_interfaces.proto#L46-L48

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #553.

### CODE REVIEW

File: wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java [hunk 1]
```diff
@@ -748,7 +748,14 @@ private String readQuotedString() {
     StringBuilder result = new StringBuilder();
     while (pos < data.length) {
       char c = data[pos++];
-      if (c == startQuote) return result.toString();
+      if (c == startQuote) {
+        // Adjacent strings are concatenated. Return if the next character is not another quote.
+        if (peekChar() != startQuote) {
```
Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java]: Can you alternate quotes ?   "foo" 'bar' "baz" ?

**Comment by swankjesse** [comment]
LGTM

File: wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java (hunk 1 above)
Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java]: I'm not certain. Will try to test.

Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java]: You can! Update.

**Pull request merged**